### PR TITLE
Delete operator bool for ErrorMessageOr<bool>

### PR DESCRIPTION
**Pull request by dimitry-**
Test: builds

### CODE REVIEW

File: src/OrbitBase/include/OrbitBase/Result.h [hunk 1]
```diff
@@ -25,7 +25,17 @@ template <typename T, typename E>
...
+class ErrorMessageOr : public Result<T, ErrorMessage> {
+ public:
+  using Result<T, ErrorMessage>::Result;
+};
+
+template <>
+class ErrorMessageOr<bool> : public Result<bool, ErrorMessage> {
+ public:
+  using Result<bool, ErrorMessage>::Result;
+  operator bool() = delete;
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/Result.h:37]: Is that also deleting the const overload? Probably. I honestly don't know.

**Review by beckerhe — APPROVED** [review]
That was fast. Thanks a lot! LGTM!

**Review by danielfenner — APPROVED** [review]
lgtm

File: src/OrbitBase/include/OrbitBase/Result.h (hunk 1 above)
Comment (dimitry-) [reply src/OrbitBase/include/OrbitBase/Result.h:37]: It is according to errors I am getting but I explicitly deleted const variant anyways

**Pull request merged**